pkg/controllers/vdb/sandbox/sandbox_controller.go

```diff
@@ -0,0 +1,180 @@
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.14.1/pkg/reconcile
+func (r *SandboxConfigMapReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := r.Log.WithValues("sandboxConfigMap", req.NamespacedName)
+	log.Info("starting reconcile of sandbox configmap")
+
+	// Fetch the ConfigMap instance
+	configMap := &corev1.ConfigMap{}
+	err := r.Get(ctx, req.NamespacedName, configMap)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// ConfigMap not found, return and don't requeue
+			log.Info("ConfigMap not found.  Ignoring since object must be deleted")
+			return ctrl.Result{}, nil
+		}
+		log.Error(err, "failed to get ConfigMap")
+		return ctrl.Result{}, err
+	}
+
+	// Fetch the VDB found in the configmap
+	vdb := &v1.VerticaDB{}
+	nm := names.GenNamespacedName(configMap, configMap.Data[verticaDBNameKey])
+	var res ctrl.Result
+	if res, err = vk8s.FetchVDB(ctx, r, configMap, nm, vdb); verrors.IsReconcileAborted(res, err) {
+		return res, err
+	}
+
+	// Iterate over each actor
+	actors := r.constructActors(vdb, log)
+	for _, act := range actors {
+		log.Info("starting actor", "name", fmt.Sprintf("%T", act))
+		res, err = act.Reconcile(ctx, &req)
+		// Error or a request to requeue will stop the reconciliation.
+		if verrors.IsReconcileAborted(res, err) {
+			log.Info("aborting reconcile of Sandbox ConfigMap", "result", res, "err", err)
+			return res, err
+		}
+	}
+	log.Info("ending reconcile of Sandbox ConfigMap", "result", res, "err", err)
+	return ctrl.Result{}, nil
+}
+
+// constructActors will a list of actors that should be run for the reconcile.
+// Order matters in that some actors depend on the successeful execution of
+// earlier ones.
+func (r *SandboxConfigMapReconciler) constructActors(_ *v1.VerticaDB,
+	_ logr.Logger) []controllers.ReconcileActor {
+	// The actors that will be applied, in sequence, to reconcile a sandbox configmap.
+	return nil
+}
+
+// predicateFuncs returns a predicate that will be used to filter
+// create/update/delete events based on specific labels
+func (r *SandboxConfigMapReconciler) predicateFuncs() predicate.Funcs {
+	return predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			// filter based on labels
+			return r.containsSandboxConfigMapLabels(e.Object.GetLabels())
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			// filter based on labels
+			return r.containsSandboxConfigMapLabels(e.ObjectNew.GetLabels())
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			// filter based on labels
+			return r.containsSandboxConfigMapLabels(e.Object.GetLabels())
+		},
+	}
+}
+
+// containsSandboxConfigMapLabels returns true if the labels map contains
+// all sandbox configmap labels
+func (r *SandboxConfigMapReconciler) containsSandboxConfigMapLabels(labels map[string]string) bool {
+	for _, label := range vmeta.SandboxConfigMapLabels {
+		_, ok := labels[label]
```

Comment:
Should we check if some of the labels have the expected values? Some of the labels we are checking are standard ones that have a constant value in our use. I'm thinking of: app.kubernetes.io/managed-by and app.kubernetes.io/component.

Reply:
I don't think that's very important it doesn't cost anything to add a check so I will do so.